processing/src/main/java/io/druid/query/groupby/epinephelinae/RowBasedGrouperHelper.java

```diff
@@ -651,6 +657,38 @@ public Object get()
     }
 
     @Override
+    public NumericColumnSelector makeMathExpressionSelector(String expression)
+    {
+      final Expr parsed = Parser.parse(expression);
+
+        final Set<String> required = Sets.newHashSet(Parser.findRequiredBindings(parsed));
+        final Map<String, Supplier<Number>> values = Maps.newHashMapWithExpectedSize(required.size());
+
+        for (final String columnName : required) {
+          values.put(
+              columnName, new Supplier<Number>()
+              {
+                @Override
+                public Number get()
+                {
+                  return row.get().getFloatMetric(columnName);
```

Comment:
@jon-wei @gianm Yep, you guys can find similar code in IncrementalIndex, too. Actually it's fixed in #2836 (of our branch) but I'll address that in here.